enhancement(host_metrics source): replace heim with sysinfo in network collector (#24818)
vectordotdev/vector: changelog.d/23646_host_metrics_network_sysinfo.enhancement.md

> The `host_metrics` source network collector now uses `sysinfo` instead of the
> unmaintained `heim` crate. `network_transmit_packets_total` is now emitted on
> all platforms (previously linux/windows only). Windows `network_transmit_packets_drop_total`
> is temporarily unavailable pending upstream sysinfo support.
> 
> authors: mushrowan

Reviewer: ```suggestion
The `host_metrics` source now emits `network_transmit_packets_total` on all platforms
(previously Linux and Windows only). Windows `network_transmit_packets_drop_total`
is temporarily unavailable pending upstream sysinfo support.

authors: mushrowan
```
